lib/puppet/indirector/report/json.rb

```diff
@@ -24,7 +24,7 @@ def save(request)
     FileUtils.mkdir_p(File.dirname(filename))
 
     begin
-      Puppet::Util.replace_file(filename, mode) do |fh|
+      Puppet::FileSystem.replace_file(filename, mode) do |fh|
```

Comment:
One subtle thing to note, `Puppet::Util.replace_file` only uses the [`default_mode` argument](https://github.com/puppetlabs/puppet/blob/de3c3a8d746bb2e2449110977a3adb58b7f7fa2a/lib/puppet/util.rb#L609-L612) when creating the file, not when updating an existing file! So if someone did:

```ini
lastrunreport = $statedir/last_run_report.yaml { mode = 0700 }
```

Then the mode is only set when the file is created or later whenever the next settings catalog is applied (when `Puppet.settings.use(:agent, ...)` is next called).

However, `Puppet::FileSystem.replace_file` **always** uses the `mode` when creating or updating files, which I think is a good thing.